README.md

```diff
@@ -1 +1,292 @@
-# node-modules-patterns
\ No newline at end of file
+# Module Patterns in Node
+
+## Overview
+
+Imagine a scenario where you have an application with 2,000 lines of code. All the code is in one file. That's a nightmare to navigate around and also difficult for different developers to collaborate without merge conflicts! What if you split the logic into 20 files which each file having 100 lines of code roughly. Much better right? So what's the best way to do that? The answer is modules!
```

Comment:
I like how this paragraph frames the problem, but it seems a little too black and white to me. There are lots of way to modularize this hypothetical code (and even some modularization strategies that put the modules in a single file).

While we don't have (or want) to acknowledge all of the ways that Node.js has overloaded the term "module," I think we can keep this framing narrative and present different files that are named appropriately as one way to approach cutting down a large code base into smaller, reusable chunks. This will also help to stave off potential confusion that `module === file`, which isn't the case.